### ISSUE TITLE
Request: Way to get all components attached in app 

### ISSUE THREAD

**Issue by laco0416** [angular/angular#7168]
Currently there is no way to get components list attached in application. To do it, we have to add some codes to constructor (or other lifecycle) of all components like below 

```ts
constructor(componentRegistry:ComponentRegistry) {
  componentRegistry.add(this);
}
```

I want a way for this to debug my application. And finally I want to lint my application about component's metadata. For example, I want to know whether there are any components which is named in camelCase. 

Is this a difficult or reasonless request?

**Comment by zoechi** [comment]
Looks like https://github.com/angular/angular/pull/7161 is what you are looking for.

**Comment by laco0416** [comment]
Thank you for telling me.

**Comment by laco0416** [comment]
Unfortunately, #7161 does not seem able to solve this.

@tbosch What do you think about this?

**Comment by laco0416** [comment]
@tbosch I'd like to hear your thinking about this. Thanks.

**Comment by laco0416** [comment]
I can walk from root component. :)

**Comment by DrewynV** [comment]
__________________________________

Hi! We just wanted to let you know we receievd your email. Thank you for submitting your request. Our aim is to get back to you within 24 hours at the latest. If you have any additional information to add to your request, please reply to this email.


Also please visit our knowledgebase (http://www.support.magicdust.com.au) to browse for further info, we are adding to it all the time.

Or here is our industry report about the 11 Essentials for a Successful Website (http://www.magicdust.com.au/web-design/.

We look forward to helping you succeed online.

Thanks

The Team @ Magicdust Digital

http://support.magicdust.com.au/

Ph (02) 9979 7474
M-F | 9 am-5:00 pm AEST (AEDT in Summer) excluding Public Holidays

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #166452.
------------------------------------------------------------------------------------------------------------


[[46d624e0506b86b7f35ec71846a979c6a5f03e5d-673073333]]

**Comment by DrewynV** [comment]
__________________________________

Hi! We just wanted to let you know we receievd your email. Thank you for submitting your request. Our aim is to get back to you within 24 hours at the latest. If you have any additional information to add to your request, please reply to this email.


Also please visit our knowledgebase (http://www.support.magicdust.com.au) to browse for further info, we are adding to it all the time.

Or here is our industry report about the 11 Essentials for a Successful Website (http://www.magicdust.com.au/web-design/.

We look forward to helping you succeed online.

Thanks

The Team @ Magicdust Digital

http://support.magicdust.com.au/

Ph (02) 9979 7474
M-F | 9 am-5:00 pm AEST (AEDT in Summer) excluding Public Holidays

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #166496.
------------------------------------------------------------------------------------------------------------


[[36342dd2ebb3cf33650bd4146e84742abad36f4e-673073815]]

**Comment by DrewynV** [comment]
__________________________________

Hi! We just wanted to let you know we receievd your email. Thank you for submitting your request. Our aim is to get back to you within 24 hours at the latest. If you have any additional information to add to your request, please reply to this email.


Also please visit our knowledgebase (http://www.support.magicdust.com.au) to browse for further info, we are adding to it all the time.

Or here is our industry report about the 11 Essentials for a Successful Website (http://www.magicdust.com.au/web-design/.

We look forward to helping you succeed online.

Thanks

The Team @ Magicdust Digital

http://support.magicdust.com.au/

Ph (02) 9979 7474
M-F | 9 am-5:00 pm AEST (AEDT in Summer) excluding Public Holidays

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #166507.
------------------------------------------------------------------------------------------------------------


[[ef1c0aba0f0f04e9ea3233660d0a3c6a26d53ce0-673073912]]

**Comment by matthewharwood** [comment]
@lacolaco this is a super old thread but did you figure out any way to do this? In my case, I've added a static method to ever component in my application and I would like to all them from an built ssr bundle:

I've figured out how to get all my routes using the  Universals ls routes: https://github.com/angular/universal/blob/fabian/ls-routes/modules/common/ls-routes/src/ls-routes.ts

But now looking for a similar feature for `ls-components`.  In the end, for each component that has `MyComponent.myStaticMethod()` i'd like to call it and collect their return values.

Long story short, do you have any leads on something similar to ls-routes but for components?
